compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -232,12 +236,20 @@ impl VirtualFileMapping {
     }
 }
 
+fn span_file_name(tcx: TyCtxt<'_>, span: Span) -> Symbol {
+    let source_file = tcx.sess.source_map().lookup_source_file(span.lo());
+    let name =
+        source_file.name.for_scope(tcx.sess, RemapPathScopeComponents::MACRO).to_string_lossy();
```

Comment:
Suggestion: `to_str().unwrap()`, we shouldn't be failing here because I would expect rust filenames to already be validated, right?

Reply:
That's probably true, but:
- If it *is* true, then there's no functional difference between `.to_str().unwrap()` and `.to_string_lossy()`.
- If it's somehow *not* true, we get an ICE, which in this context seems worse than emitting a few `�` characters in the coverage metadata.

(Also note that this code is just being moved, so `.to_string_lossy()` is already the current behaviour, and I didn't see a compelling reason to change it. Eventually I plan to stop using `Symbol` entirely, but that's beyond the scope of this PR.)

Reply:
Alright that's fine